File: src/main/java/space/maxkonkin/nasapicbot/config/TelegramConfig.java
package space.maxkonkin.nasapicbot.config;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.experimental.FieldDefaults;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.PropertySource;

@Configuration
@PropertySource(value = "classpath:application-${SPRING_PROFILE}.yaml", factory = YamlPropertySourceFactory.class)
@Getter
@FieldDefaults(level = AccessLevel.PRIVATE)
public class TelegramConfig {
    @Value("${telegram.bot-path}")
    String botPath;
    @Value("${telegram.bot-token}")
    String botToken;
    @Value("${telegram.webhook-path}")
    String webhookPath;
    @Value("${telegram.bot-name}")
    String botName;
    @Value("${message.errorText.text}")
    String errorText;
}

File: src/main/kotlin/space/maxkonkin/nasapicbot/config/TelegramConfig.kt
package space.maxkonkin.nasapicbot.config

import org.telegram.telegrambots.meta.api.objects.commands.BotCommand

fun loadTelegramConfig(properties: Map<String, Any>): TelegramConfig {

    return TelegramConfig(
        botPath = requireNotNull((properties["telegram"] as Map<*, *>)["bot-path"]) as String,
        botToken = System.getenv("BOT_TOKEN"),
        webhookPath = requireNotNull((properties["telegram"] as Map<*, *>)["webhook-path"]) as String,
        botName = requireNotNull((properties["telegram"] as Map<*, *>)["bot-name"]) as String,
        errorText = requireNotNull((properties["message"] as Map<*, *>)["error-text"]) as String,
        listOfCommands = listOfCommands
    )
}

data class TelegramConfig(
    val botPath: String,
    val botToken: String,
    val webhookPath: String,
    val botName: String,
    val errorText: String,
    val listOfCommands: List<BotCommand>
)

private val listOfCommands = listOf(
    BotCommand("/start", "Получить описание"),
    BotCommand("/help", "Получить описание"),
    BotCommand("/today", "Скинуть сегодняшнюю картинку"),
    BotCommand("/random", "Скинуть случайную картинку"),
    BotCommand("/schedule", "Переключить отправку сегодняшней картинки по расписанию"),
    BotCommand("/translate", "Переключить перевод EN->RU"),
)